packages/shared/src/permissions/permissions.utils.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  Permission,
+  UserPermission,
+  UserPermissions,
+} from "back-end/types/organization";
+
+export function hasPermission(
+  userPermissions: UserPermissions | undefined,
+  permissionToCheck: Permission,
+  project?: string | undefined,
+  envs?: string[]
+): boolean {
+  if (!userPermissions) {
+    return false;
+  }
```

Comment:
Make sure this still works for the REST api.  We have 2 types of keys - Personal Access Tokens with a userId and secret tokens with a role.

Reply:
I ran a few tests this morning via Postman, and I was able to confirm that if the user was an admin and used their personal access token to create an experiment, it was created successfully, and if a collaborator tried the same with their personal access token, it returned a permission error.

<img width="1161" alt="Screen Shot 2023-08-17 at 9 24 15 AM" src="https://github.com/growthbook/growthbook/assets/75274610/e1666246-e916-4baa-b414-19b02a293cf2">

I did notice one use case, which I think is expected, but if the collaborator used the org's secret token, they were able to create an experiment, despite their user not having permission to do so.